### PR TITLE
pythonPackages.ovito: init at 3.0.0

### PR DESCRIPTION
**Pull request by costrouc**
###### Motivation for this change

ovito is a visualization package for visualizing molecules and doing quick analysis.

###### Things done

pythonPackages.ovito: init at 3.0.0

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by costrouc** [comment]
Currently builds but I am running into a QT error when trying to run the binary.

```
/nix/store/2xgw34ydjhkdhq18wd448qprm4j6nwrf-ovito-3.0.0/bin $ ./ovito
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

aborted (core dumped)
```

Any help would be much appreciated!

**Comment by costrouc** [comment]
Tried to run it in a pure nix-shell environment however it still segfaulted...

```
let pkgs = import <nixpkgs> { config = { }; };
in
pkgs.mkShell {
  buildInputs = [ pkgs.python36Packages.ovito ];

  shellHook = ''
    export QT_PLUGIN_PATH=${pkgs.qt5.qtbase.bin}/${pkgs.qt5.qtbase.qtPluginPrefix}
  '';
}
```

**Comment by costrouc** [comment]
Fixed and ready to merge. I have tested that the qt5 application works in NixOS. The QT5 documentation on nixpkgs was invaluable for fixing the issue.

**Comment by costrouc** [comment]
@GrahamcOfBorg build python36Packages.ovito

**Comment by costrouc** [comment]
This package takes a long time to build but I can promise that it works on x86_64-linux

**Comment by costrouc** [comment]
Bump. Ready for merge. Build takes over an hour and I can verify that the qt application works.

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/chemistry/ovito/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchgit
+, pkgs, cmake
```
[the remark this answers is not in the document]
Comment (costrouc) [reply pkgs/applications/science/chemistry/ovito/default.nix]: Fixed. Turns out `pkgs` was not being using in the derivation.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -422,6 +422,10 @@ in {
 
   outcome = callPackage ../development/python-modules/outcome {};
 
+  ovito = pkgs.libsForQt5.callPackage ../applications/science/chemistry/ovito {
```
[the remark this answers is not in the document]
Comment (costrouc) [reply pkgs/top-level/python-packages.nix]: Fixed. Added `toPythonModule` to derivation.

**Comment by costrouc** [comment]
Fixed. Move to `python-modules`, added `toPythonModule`, and removed `pkgs` from derivation.

**Comment by FRidh** [comment]
I just has a look at the source code, and they vendor a lot. It's likely a lot of work to replace their vendored dependencies with our Nixpkgs builds, however, I don't think it is reasonable to let Hydra build this package.

**Comment by costrouc** [comment]
What if I were to patchelf the binary that they provide? https://ovito.org/index.php/download

Not ideal but would make the build trivial.

**Comment by FRidh** [comment]
Not sure if it would actually work well, considering also the Python dependencies.
In any case, you can have it like this but set `platforms = []` so Hydra won't build it.

**Comment by FRidh** [comment]
Ah, wait, it was `hydraPlatforms`. See the Nixpkgs manual.

**Comment by costrouc** [comment]
Makes more sense thanks for showing me that. Fixed with `hydraPlayforms = [ ];'

**Comment by costrouc** [comment]
Bump. Ready for merge. I have added `hydraPlatforms = [ ]`. No actual builds will be done on hydra.

**Pull request merged**